firedrake/mesh.py

```diff
@@ -337,8 +338,288 @@ def _from_cell_list(dim, cells, coords, comm):
     return plex
 
 
-class MeshTopology(object):
-    """A representation of mesh topology."""
+class AbstractMeshTopology(object, metaclass=abc.ABCMeta):
+    """A representation of an abstract mesh topology without a concrete
+        PETSc DM implementation"""
+
+    def __init__(self, name):
+        """Initialise an abstract mesh topology.
+
+        :arg name: name of the mesh
+        """
+
+        utils._init()
+
+        self.name = name
+
+        # A cache of shared function space data on this mesh
+        self._shared_data_cache = defaultdict(dict)
+
+        # Cell subsets for integration over subregions
+        self._subsets = {}
+
+        self._grown_halos = False
+
+        # A set of weakrefs to meshes that are explicitly labelled as being
+        # parallel-compatible for interpolation/projection/supermeshing
+        # To set, do e.g.
+        # target_mesh._parallel_compatible = {weakref.ref(source_mesh)}
+        self._parallel_compatible = None
+
+        def callback(self):
+            """Finish initialisation."""
+            raise NotImplementedError("Should be defined in a subclass")
```

Comment:
This is not referenced anywhere, so just remove it.

Reply:
done